### ISSUE TITLE
Location of config.xml?

### ISSUE THREAD

**Issue by townsmcp** [1activegeek/docker-airconnect#27]
Hi all,
Sorry to be dumb, but where is the config.xml file stored? I want to edit it and set some players to not show up but I just cant seem to find the config file

**Comment by 1activegeek** [comment]
Please take a look at a few of the other issues here that are closed such as #20 - these should point you in the right direction. Short answer - you have to specify the path when you mapped the file.

**Comment by townsmcp** [comment]
Not sure if anyone would be interested, however I sorted the problem out.
I wanted to be able to only map non AirPlay Sonos devices. I have a mixture of AirPlay Sonos devices and non Airplay Sonos devices. I didnt want AirConnect to be mapping out AirPlay enabled Sonos devices (why have a device mapped twice?).
I use Portainer so editing the Docker config etc is much easier than CLI (highly recommend Docker or some other GUI based setup) for configuring Docker.

There are 2 ways to achieve this:

- Set the environmental variable to only scan and publish certain player types.
-- This can be achived by editing the container, and adding a new env varaible AIRUPNP_VAR and adding the relevant values eg -o S1,S3,S5,S9,S12,ZP80,ZP90,S15,ZP100,ZP120 (I found the values [here](https://github.com/philippe44/AirConnect/issues/316))
![image](https://user-images.githubusercontent.com/14061617/134791730-43b3a037-2c7c-4d78-a955-74bbfb5b4bcd.png)

- Creating a config.xml file from the current config. Save the file to a persistent mounted volume, edit the file from the host and republish the Docker container:
-- First, create a new directory on the host, eg /opt/airupnp/config
-- Mount the volume for easy editing in the host eg using nano. This should point to the any location in the container however there is already a /config directory
![image](https://user-images.githubusercontent.com/14061617/134791916-1f97602a-313c-4f4b-9434-dbca6ce43170.png)
--- Deploy the container with the new config
--- With the running AirConenct container, go to a console
--- Change directory to /bin
--- Run the following command. This will take the current config and output it to a config.xml file
```
airupnp-x86-64 -i config.xml
```
![image](https://user-images.githubusercontent.com/14061617/134792004-933023b8-3a62-4687-a203-73aba3a6e12f.png)
--- Copy this newly created config.xml to /config/airupnp
```
cp config.xml /config/config.xml
```
---From your host, edit the file
```
nano /opt/airupnp/config/config.xml
```
--- In the file, you can then turn a player on and off from being published by changing <enabled>1</enabled> to <enabled>0</enabled>
--- Save the file
--- In the ENV section of the container, set AIRUPNP_VAR to -x /config/airupnp/config.xml
![image](https://user-images.githubusercontent.com/14061617/134792118-3d9f1d1e-343e-4936-9b9f-372e3542aa90.png)
--- Redeploy the container



So by default, there is no config.xml file - AirConnect performs a scan every time it starts up and uses those scan results to publish devices to AirPlay.
To control the publication of non Airplay devices

As a side note, the same works for Chromecast devices however the env variabe name is set to `AIRCAST_VAR` and the command to export the config to file is `aircast-x86-64 -i config.xml`
